services/webnn/ort/graph_builder_ort.cc

```diff
@@ -733,6 +736,146 @@ GraphBuilderOrt::AddInstanceNormalizationOperation(
   return base::ok();
 }
 
+[[nodiscard]] base::expected<void, mojom::ErrorPtr>
+GraphBuilderOrt::AddLayerNormalizationOperation(
+    const mojom::LayerNormalization& layer_normalization) {
+  const OperandDataType input_data_type =
+      GetOperand(layer_normalization.input_operand_id).descriptor.data_type();
+
+  const std::string input_name =
+      GetOperandName(layer_normalization.input_operand_id);
+  std::vector<const char*> input_names = {input_name.c_str()};
+  const std::vector<uint32_t>& input_shape =
+      GetOperand(layer_normalization.input_operand_id).descriptor.shape();
+
+  const std::string node_name = GetNodeName(layer_normalization.label);
+  const std::string output_name =
+      GetOperandName(layer_normalization.output_operand_id);
+  std::array<const char*, 1> output_names = {output_name.c_str()};
+
+  auto axes = layer_normalization.axes;
+  // ONNX doesn't support empty axes. When axes is empty, the mean equals to
+  // input, output = bias + (scale * 0)
+  if (axes.empty()) {
+    if (layer_normalization.bias_operand_id) {
+      base::CheckedNumeric<uint32_t> checked_input_size =
+          std::accumulate(input_shape.begin(), input_shape.end(),
+                          base::CheckedNumeric<uint32_t>(1), std::multiplies());
+      if (!checked_input_size.IsValid()) {
+        return NewNotSupportedError("The size of input is too large.");
+      }
+      std::vector<float> zero_data(checked_input_size.ValueOrDie(), 0.0f);
+      std::string zero_name;
+      switch (input_data_type) {
+        case OperandDataType::kFloat16: {
+          std::vector<uint16_t> zero_data_fp16 =
```

Comment:
Can we construct `zero_data_fp16` directly? i.e.
```suggestion
          std::vector<uint16_t> zero_data_fp16(checked_input_size.ValueOrDie(), fp16_ieee_from_fp32_value(0.0f));
```

Reply:
Yes, you are right!